crates/cheatcodes/src/lib.rs

```diff
@@ -69,10 +71,25 @@ pub(crate) trait Cheatcode: CheatcodeDef + DynCheatcode {
         self.apply(ccx.state)
     }
 
+    /// Applies this cheatcode to the given context and executor.
+    ///
+    /// Implement this function if you need access to the executor.
+    fn apply_full_with_executor<DB: DatabaseExt, E: CheatcodesExecutor>(
```

Comment:
`#[inline(always)]` and naming can be improved here, maybe `apply_full -> apply_stateful`, and this to `apply_full`

Reply:
bump cc @mattsse

Reply:
ah yep, renamed in 5f31795